test/e2e/e2e_test.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/types"
+
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1alpha2"
+	"sigs.k8s.io/kueue/test/e2e/framework"
+
+	"sigs.k8s.io/kueue/pkg/util/testing"
+)
+
+// +kubebuilder:docs-gen:collapse=Imports
+
+var _ = ginkgo.Describe("End To End Suite", func() {
+	var ns *corev1.Namespace
+	ginkgo.BeforeEach(func() {
+		ns = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "e2e-",
+			},
+		}
+		gomega.Expect(k8sClient.Create(ctx, ns)).To(gomega.Succeed())
+	})
+	ginkgo.AfterEach(func() {
+		gomega.Expect(k8sClient.Delete(ctx, ns)).To(gomega.Succeed())
+	})
+	ginkgo.When("Creating a Job Without Queueing", func() {
+		ginkgo.It("Should stay in suspended", func() {
+			sampleJob := testing.MakeJob("test-job", ns.Name)
+			annotations := map[string]string{
+				"kueue.x-k8s.io/queue-name": "main",
+			}
+			sampleJob.ObjectMeta.Annotations = annotations
+			gomega.Expect(k8sClient.Create(ctx, &sampleJob.Job)).Should(gomega.Succeed())
+			lookupKey := types.NamespacedName{Name: "test-job", Namespace: ns.Name}
+			createdJob := &batchv1.Job{}
+			gomega.Eventually(func() bool {
+				if err := k8sClient.Get(ctx, lookupKey, createdJob); err != nil {
+					return false
+				}
+				return *createdJob.Spec.Suspend
+			}, framework.Timeout, framework.Interval).Should(gomega.BeTrue())
+			gomega.Expect(k8sClient.Delete(ctx, &sampleJob.Job)).Should(gomega.Succeed())
```

Comment:
worth checking that there is a corresponding workload object that is not admitted.